packages/react/src/VideoPlayer/VideoPlayer.tsx

```diff
@@ -0,0 +1,189 @@
+  ref?: React.RefObject<HTMLVideoElement>
+} & React.HTMLProps<HTMLVideoElement>
+
+const Root = ({
+  poster,
+  title,
+  branding = true,
+  children,
+  className,
+  onPlay,
+  onPause,
+  onLoadedMetadata,
+  onPlaying,
+  ref,
+  ...rest
+}: VideoPlayerProps) => {
+  const videoPlayerRef = useRef<HTMLVideoElement>(null)
+  const videoWrapperRef = useRef<HTMLDivElement>(null)
+  const videoRef = ref ? ref : videoPlayerRef
+  const [playing, setPlaying] = useState(false)
+  const [playedTime, setPlayedTime] = useState(0)
+  const [totalTime, setTotalTime] = useState(0)
+  const [trackInformation, setTrackInformation] = useState<TextTrackCueList | undefined>(undefined)
+
+  /* > Play / Pause                                                    */
+  const handleVideoPlayback = () => {
+    if (videoRef.current) {
+      if (videoRef.current.paused) {
+        videoRef.current.play()
+      } else {
+        videoRef.current.pause()
+      }
+    }
+  }
+
+  /* > Hide Default Captions                                                   */
+  useEffect(() => {
+    if (videoRef.current) {
+      for (let i = 0; i < videoRef.current.textTracks.length; i++) {
+        videoRef.current.textTracks[0].mode = 'hidden'
+      }
+    }
+  }, [videoRef])
+
+  /* > Set breakpoint classes                                                   */
+
+  useEffect(() => {
+    const handleResize = () => {
+      const breakpoint = videoWrapperRef.current?.getBoundingClientRect().width
+      if (breakpoint && breakpoint < 650) {
+        videoWrapperRef.current.classList.add(styles.VideoPlayer__breakpointSmall)
+      } else {
+        videoWrapperRef.current?.classList.remove(styles.VideoPlayer__breakpointSmall)
+      }
+    }
+
+    const resizeObserver = new ResizeObserver(_ => {
+      if (videoWrapperRef.current) {
+        handleResize()
+      }
+    })
+
+    handleResize()
+
+    const currentRef = videoWrapperRef.current
+
+    resizeObserver.observe(videoWrapperRef.current as Element)
+    return () => {
+      resizeObserver.unobserve(currentRef as Element)
+    }
+  }, [videoWrapperRef, playing])
+
+  return (
+    <div
+      className={clsx(
+        styles.VideoPlayer__container,
+        styles.VideoPlayer__overlays,
+        !playing && styles.VideoPlayer__showOverlays,
+      )}
+      ref={videoWrapperRef}
+    >
+      {/* eslint-disable-next-line jsx-a11y/media-has-caption */}
+      <video
+        ref={videoRef}
+        title={title}
+        controls={false}
+        poster={poster}
+        {...rest}
+        className={clsx(styles.VideoPlayer, className)}
+        onPlay={e => {
+          setPlaying(true)
+          onPlay && onPlay(e)
+        }}
+        onPause={e => {
+          setPlaying(false)
+          onPause && onPause(e)
+        }}
+        onPlaying={e => {
+          videoRef.current?.textTracks[0].cues && setTrackInformation(videoRef.current.textTracks[0].cues)
+          onPlaying && onPlaying(e)
+        }}
```

Comment:
Performance optimisation: Could you hoist this function to outside of the render function, so that we're not recreating it each time. You can wrap it inside a `useCallback` to memoize it after. Example from [another component](https://github.com/primer/brand/blob/551c47cf635454a5eb82b3cca1ef833e2f53c1b4/packages/react/src/ActionMenu/ActionMenu.tsx#L124).

Reply:
Could we also please apply this optimisation to all other inline function declarations?

Reply:
This has been updated in the following [SHA](https://github.com/primer/brand/pull/373/commits/6f9a9c8740844ac6d50be622f1c9aab106c564dc)